include/mata/nfa.hh

```diff
@@ -153,11 +153,18 @@ struct Move {
     Move(Symbol symbolOnTransition, const StateSet& states_to) :
             symbol(symbolOnTransition), targets(states_to) {}
 
-    //Trying to make Move movable, does it make sense?
-    Move(Move&& rhs) = default;
+    Move(Move&& rhs) noexcept : symbol{ rhs.symbol }, targets{ std::move(rhs.targets) } {
+        rhs.targets.clear();
```

Comment:
Do we need to clear targets of `rhs`? Shouldn't `std::move(rhs.targets)` do that?

Reply:
Targets is a `OrdVector`, which has its own implementation of move operations. In this case, yeah, the custom implementation of move operations in `OrdVector` will clear targets. However, if we were to change the `OrdVector` for, for example, `std::map`, we would not get the same implementation and the `targets` would remain in valid, yet unknown state as per C++ standard for move operations. This ensures that this cannot happen, and it will always be a valid, yet unknown state, which is known in this case: the default empty state, even if you are supposed to never use the moved out value in said variable until a new value replaces this value state.

We could optimize like you suggest. I am not sure if it is the better approach here, though. But I agree that it will work as required as well.

Reply:
Ok, I am fine with clearing it.

Reply:
I decided to implement your suggestion. The state of the `other`/`rhs` object is always valid. Hence, the code should be a little faster, yet still correct.